src/workerd/api/blob.c++

```diff
@@ -143,13 +143,23 @@ public:
       : unread(blob->data),
         blob(kj::mv(blob)) {}
 
+  // Attempt to read a maximum of maxBytes from the remaining unread content of the blob
+  // into the given buffer. It is the caller's responsibility to ensure that buffer has
+  // enough capacity for at least maxBytes.
+  // The minBytes argument is ignored in this implementation of tryRead.
+  // The buffer must be kept alive by the caller until the returned promise is fulfilled.
+  // The returned promise is fulfilled with the actual number of bytes read.
   kj::Promise<size_t> tryRead(void* buffer, size_t minBytes, size_t maxBytes) override {
     size_t amount = kj::min(maxBytes, unread.size());
-    memcpy(buffer, unread.begin(), amount);
-    unread = unread.slice(amount, unread.size());
+    if (amount > 0) {
```

Comment:
Nitpick: I doubt this branch improves performance (the impact is probably negligible) but it does make the code more complex, so I'd tend to lean against it, though I don't feel strongly. OTOH, maybe it fixes a UBSAN issue -- if so then it makes sense.

Reply:
Yeah, it's minor which is why I stuck it in a separate commit that's easy to drop if necessary. If I recall correctly, I *think* technically it would only be UB if `buffer` was `const` and most compilers (all?) would treat `amount = 0` as a non-op but my memory there is pretty fuzzy. That said, I tend to prefer being explicit about it and just skipping both the `memcpy` and `slice` when it's obviously that we just don't *need* to.